Neos.ContentRepository.Core/Classes/Infrastructure/PostgreSQLSchemaService.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Infrastructure;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Column;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SchemaException;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeAggregateId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * Provide postgresql specific doctrine DBAL column schema definitions for common types in the content repository to
+ * produce consistent columns across projections.
+ *
+ * @internal Because we might need to change the input and output format of functions within.
+ */
+final class PostgreSQLSchemaService implements SchemaServiceInterface
+{
+    private function __construct()
+    {
+    }
+
+    public static function create(): self
+    {
+        return new self();
+    }
+
+    /**
+     * The NodeAggregateId is limited to 64 ascii characters and therefore we should do the same in the database.
+     *
+     * @see NodeAggregateId
+     */
+    public function columnForNodeAggregateId(string $columnName): Column
+    {
+        return (new Column($columnName, Type::getType(Types::STRING)))
+            ->setLength(64);
+    }
+
+    /**
+     * The ContentStreamId is generally a UUID, therefore not a real "string" but at the moment a strified identifier,
+     * we can safely store it as binary string as the charset doesn't matter
+     * for the characters we use (they are all asscii).
+     *
+     * We should however reduce the allowed size to 36 like suggested
+     * here in the schema and as further improvement store the UUID in a more DB friendly format.
+     *
+     * @see ContentStreamId
+     */
```

Comment:
it seems they are the same comments like in the interface? i would just like to remove them all in this class then. Comments get outdated so easily and duplicating them is even worse (especially if its internal and no one reads them here ^^ )

```suggestion
    /**
     * The ContentStreamId is generally a UUID, therefore not a real "string" but at the moment a strified identifier,
     * we can safely store it as binary string as the charset doesn't matter
     * for the characters we use (they are all asscii).
     *
     * We should however reduce the allowed size to 36 like suggested
     * here in the schema and as further improvement store the UUID in a more DB friendly format.
     *
     * @see ContentStreamId
     */
```